bundle/phases/deploy.go

```diff
@@ -124,60 +126,89 @@ is removed from the catalog, but the underlying files are not deleted:`
 	return approved, nil
 }
 
-// The deploy phase deploys artifacts and resources.
-func Deploy(outputHandler sync.OutputHandler) bundle.Mutator {
+func deployCore(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 	// Core mutators that CRUD resources and modify deployment state. These
 	// mutators need informed consent if they are potentially destructive.
-	deployCore := bundle.Defer(
-		bundle.Seq(
-			bundle.LogString("Deploying resources..."),
-			terraform.Apply(),
-		),
-		bundle.Seq(
-			terraform.StatePush(),
-			terraform.Load(),
-			apps.InterpolateVariables(),
-			apps.UploadConfig(),
-			metadata.Compute(),
-			metadata.Upload(),
-			bundle.LogString("Deployment complete!"),
-		),
-	)
+	cmdio.LogString(ctx, "Deploying resources...")
+	diags := bundle.Apply(ctx, b, terraform.Apply())
+
+	// following original logic, continuing with sequence below even if terraform had errors
```

Comment:
This is expected; TF may error after successfully creating resources, so the resulting state always needs to be pushed back to the workspace.